plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -60,7 +60,7 @@
    * Called by OSGi Declarative Services Runtime when the {@link GoogleLoginService} is activated
    * as an OSGi service.
    */
-  protected void activate(ComponentContext context) {
+  protected void activate() {
     IWorkbench workbench = PlatformUI.getWorkbench();
     IEclipseContext eclipseContext = workbench.getService(IEclipseContext.class);
     IShellProvider shellProvider = new SameShellProvider(workbench.getDisplay().getActiveShell());
```

Comment:
Oops, that's not what you want: that will fix the service with the shell of the active window / dialog _at creation time_.  I think you want:
```
IShellProvider shellProvider = new IShellProvider() {
  public Shell getShell() {
    return workbench.getDisplay().getActiveShell();
  }
}
```